spec/hitnmiss/file_driver_spec.rb

```diff
@@ -0,0 +1,196 @@
+require 'spec_helper'
+
+describe "Hitnmiss::FileDriver" do
+  let(:folder) { File.expand_path("../../support/cache", __FILE__) }
+  subject { Hitnmiss::FileDriver.new(folder) }
+
+  before do
+    subject
+  end
+
+  after do
+    Dir["#{folder}/*"].each { |filename| File.unlink(filename) }
+  end
+
+  describe ".new" do
+    it "sets ths folder to the passed in folder" do
+      folder = subject.instance_variable_get(:@folder)
+      expect(folder).to match("support/cache")
+    end
+  end
+
+  describe "#set" do
+    context 'when given entity has an expiration' do
+      it "caches the given value, using the given key and expiration" do
+        entity = Hitnmiss::Entity.new('some_value', expiration: 1)
+        now = Time.now.utc
+        Timecop.freeze(now) do
+          subject.set('some_key', entity)
+        end
+        cache = JSON.parse(File.read(File.join(folder, "some_key")))
+        expect(cache['value']).to eq('some_value')
+        expect(cache['expiration']).to eq(now.to_i + 1)
+      end
+    end
+
+    context 'when given entity has a fingerprint' do
+      it 'caches the given value with key, expiration, and fingerprint' do
+        entity = Hitnmiss::Entity.new('some_value', expiration: 1,
+                                      fingerprint: 'foofingerprint')
+        now = Time.now.utc
+        Timecop.freeze(now) do
+          subject.set('some_key', entity)
+        end
+        cache = JSON.parse(File.read(File.join(folder, "some_key")))
+        expect(cache['value']).to eq('some_value')
+        expect(cache['expiration']).to eq(now.to_i + 1)
+        expect(cache['fingerprint']).to eq('foofingerprint')
+      end
+    end
+
+    context 'when given entity has a last_modified' do
+      it 'caches the given value with key, expiration, and last_modified' do
+        entity = Hitnmiss::Entity.new('some_value', expiration: 1,
+                                      fingerprint: 'foofingerprint',
```

Comment:
did you intend for `fingerprint` to be set here? It looks like from the description you don't care.